persistence-modules/java-jpa-3/src/main/java/com/baeldung/jpa/equality/EqualById.java

```diff
@@ -0,0 +1,54 @@
+package com.baeldung.jpa.equality;
+
+import javax.persistence.*;
+
+@Entity
+public class EqualById {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+
+    private String email;
+
+    public EqualById() {
+    }
+
+    public Long getId() {
+        return id;
+    }
+
+    public void setId(Long id) {
+        this.id = id;
+    }
+
+    public String getEmail() {
+        return email;
+    }
+
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((id == null) ? 0 : id.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (obj instanceof EqualById)
+            return ((EqualById) obj).getId() == getId();
```

Comment:
For wrapper classes, the == operator would only return true for values in the range -127 to 127. Should use the equals method to compare the ids.

Reply:
Fixed